src/options/options.js

```diff
@@ -97,22 +97,78 @@ document.addEventListener("alpine:init", () => {
       );
       return result;
     },
+    /**
+     * Break array of repositories into chunks of 100, and
+     * store them independently in Chrome storage
+     *
+     * @param {String} workspaceSlug the slug of the Orbit workspace, used to generate a key for the repositories
+     * @param {Array<String>} repositories the un-chunked array of repository names
+     *
+     * @returns Array<String> the addresses of the chunked repositories
+     */
+    _persistRepositories(workspaceSlug, repositories) {
+      // Addresses of the chunked repositories so we can retrieve them later
+      let repository_keys = [];
+      let counter = 1;
+
+      // If there are 240 repositories for workspace "sally", they will be saved as:
+      //
+      // sally:repositories:1 // Repositories 0-100
+      // sally:repositories:2 // Repositories 101-200
+      // sally:repositories:3 // Repositories 201-240
+      while (repositories.length > 0) {
+        let key = `${workspaceSlug}:repositories:${counter}`;
+
+        repository_keys.push(key);
+        chrome.storage.sync.set({
+          [key]: repositories.splice(0, 100),
+        });
+        counter++;
+      }
+
+      return repository_keys;
+    },
+    /**
+     * Build the object of data to persist, including:
+     * - token: the users API token
+     * - workspace: slug of the selected workspace
+     * - repository_keys: addresses of the chunked repositories
+     *
+     * @param {String} token the users API token
+     * @param {String} workspaceSlug the slug of the Orbit workspace, used to generate a key for the repositories
+     * @param {Array<String>} repositories the un-chunked array of repository names
+     *
+     * @returns {token, workspace, repository_keys}
+     */
+    _buildStorageObject(token, workspaceSlug, repositories) {
+      const storageObject = {};
+
+      // Set common variables
+      storageObject["token"] = token;
+      storageObject["workspace"] = workspaceSlug;
+      storageObject["repository_keys"] = this._persistRepositories(
```

Comment:
I would recommend breaking this line into two steps:

```javascript
let repositoryKeys = async this._persistRepositories(…)

// ...

storageObject["repository_keys"] = repositoryKeys
```

It’s functionally identical, but iI find it easier to follow (as it’s now obvious that `_persistRepositories` returns a list of keys)